src/main/java/org/candlepin/pki/certs/UeberCertificateGenerator.java

```diff
@@ -109,22 +116,21 @@ public UeberCertificate generate(String ownerKey, Principal principal) {
         try {
             // There can only be one ueber certificate per owner, so delete the existing and regenerate it.
             this.ueberCertCurator.deleteForOwner(owner);
-            return  this.generateUeberCert(owner, principal.getUsername());
+            return  this.generateUeberCert(owner, username);
         }
         catch (Exception e) {
-            log.error("Problem generating ueber cert for owner: {}", ownerKey, e);
-            throw new BadRequestException(i18n.tr("Problem generating ueber cert for owner {0}", ownerKey),
-                e);
+            throw new BadRequestException(i18n.tr(
```

Comment:
Can we add back the log.error before throwing this? It is useful to have the details/stacktrace of why exactly the cert generation failed in the logs. If we only throw an HTTP-level exception then that would be lost